src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -858,7 +858,7 @@ private InferenceResult FixDependentParameters(ref CompoundUseSiteInfo<AssemblyS
             // dependent on anything. We need to first determine which parameters need to be 
             // fixed, and then fix them all at once.
 
-            var needsFixing = new bool[_methodTypeParameters.Length];
+            Span<bool> needsFixing = stackalloc bool[_methodTypeParameters.Length];
```

Comment:
>stackalloc bool[_methodTypeParameters.Length]; [](http://example.com/codeflow?start=37&length=46)

Why is this a right change? There could be a lot of type parameters. Perhaps the logic should be conditional on some threshold length?

Reply:
I agree. This is a "most cases this is fine but customers do bad things" scenario. We need some threshold in this allocation.

Reply:
i am literally reminded of the native IDE blowing the stack for this exact reason, and we had to do a DTS because of it.  Why?  Because user liked lots of type-parameters, even up to the runtime limit (which i think was like 2^16).  It was *awful*.

Reply:
⏱️ Sure, this will be easy. I'll set the threshold at 1024 based on a comment by @stephentoub in another thread.

Reply:
➡️ Now uses `stackalloc bool[]` for 0-1024 parameters, and `new bool[]` for more than that.

Reply:
> based on a comment in another thread.

What thread? And 1024 feels unnecessary high to me.

Reply:
https://github.com/dotnet/roslyn/pull/67999#discussion_r1179651552

Note that it only allocates 1KiB on the stack for the specific case of a method having exactly 1024 parameters. If the method has <10 parameters, it only puts <10 bytes on the stack.

Reply:
> Note that it only allocates 1KiB on the stack for the specific case of a method having exactly 1024 parameters. If the method has <10 parameters, it only puts <10 bytes on the stack.

That is understood. 10 looks like a good enough number to me. It is really not common to have more than 10 type parameters on a method, so trying to optimize it with the risk of blowing out the stack is not worth it, in my opinion.

Reply:
⏱️ Will set to value 10 before preparing to merge

Reply:
Was there a reason that BitVector doesn't work in this scenario? It feels like that would be pretty good for automatically doing stack-only, with minimal space usage, for smaller numbers of type parameters, and heap for large numbers of type parameters.

Reply:
➡️ This code has been changed to use `BitVector` in the main branch, so I removed this change from this pull request